edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -190,10 +190,12 @@ public static async Task<int> MainAsync(IConfiguration configuration)
                         uint persistentVolumeClaimDefaultSizeMb = configuration.GetValue<uint>(K8sConstants.PersistentVolumeClaimDefaultSizeInMbKey);
                         string deviceNamespace = configuration.GetValue<string>(K8sConstants.K8sNamespaceKey);
                         var kubernetesExperimentalFeatures = KubernetesExperimentalFeatures.Create(configuration.GetSection("experimentalFeatures"), logger);
-                        string edgeK8sObjectOwnerApiVersion = configuration.GetValue<string>(K8sConstants.EdgeK8sObjectOwnerApiVersionKey);
-                        string edgeK8sObjectOwnerKind = configuration.GetValue<string>(K8sConstants.EdgeK8sObjectOwnerKindKey);
-                        string edgeK8sObjectOwnerName = configuration.GetValue<string>(K8sConstants.EdgeK8sObjectOwnerNameKey);
-                        string edgeK8sObjectOwnerUid = configuration.GetValue<string>(K8sConstants.EdgeK8sObjectOwnerUidKey);
+                        var moduleOwner = new KubernetesModuleOwner(
+                            configuration.GetValue<string>(K8sConstants.EdgeK8sObjectOwnerApiVersionKey),
+                            configuration.GetValue<string>(K8sConstants.EdgeK8sObjectOwnerKindKey),
+                            configuration.GetValue<string>(K8sConstants.EdgeK8sObjectOwnerNameKey),
+                            configuration.GetValue<string>(K8sConstants.EdgeK8sObjectOwnerUidKey));
+                        bool runAsNonRoot = configuration.GetValue<bool>(K8sConstants.RunAsNonRootKey);
```

Comment:
GetValue() returns default(bool) if key is not present?

Reply:
yep